client/src/components/TrackList.vue

```diff
@@ -46,7 +46,8 @@ export default defineComponent({
 
   components: { TrackItem },
 
-  setup(props, { emit }) {
+  setup(props, { emit, root }) {
+    const prompt = root.$prompt;
```

Comment:
You can't use the prompt here.  TrackList is in the distributed library, but the prompt isn't. anything in `src/` cannot use prompt.

I think it'd be nice to break this functionality out of the `TrackList` anway.  What do you think about a slot down where the tooltip is? Your delete widget can live inside `app/`.  You don't even need to pass `virtualListItems` as a prop, since it's just a computed property of checkedTrackIds.  This delete button could be a completely self-contained thing.

```
const checkedTrackIdsRef = useCheckedTrackIds();
```

Another alternative is to just use a regular `v-dialog`.

Reply:
FWIW, we can also move the prompt service into the distributable, but I didn't do that for fear that downstreams would use it and then we'd be stuck supporting the prompt.  I generally dislike the prompt, but it's convenient and not a high priority to change.

A fourth option is to rewrite the prompt to be a composition function so you can `let prompt = usePrompt()`;

Then we could move it into the distributed bundle without having to expose it as part of the official api.   I actually kinda dig that option.  So before you go rewriting a bunch of stuff, LMK what options you prefer.